schema.json

```diff
@@ -1,255 +1,266 @@
-        },
-        {
-          "properties": {
-            "type": {
-              "enum": ["bool"]
-            },
-            "name": {
-              "$ref": "#/definitions/name"
-            }
-          },
-          "required": ["name", "type"]
-        },
-        {
-          "properties": {
-            "type": {
-              "enum": ["int8", "int16", "int32", "int64",
-                       "uint8", "uint16", "uint32", "uint64"]
-            },
-            "name": {
-              "$ref": "#/definitions/name"
-            },
-            "labels": {
-              "type": "array",
-              "items": {
-                "$ref": "#/definitions/label"
-              }
-            },
-            "prefix": {
-              "type": "integer",
-              "maximum": 255,
-              "minimum": 0
-            },
-            "ranges": {
-              "type": "array",
-              "items": {
-                "$ref": "#/definitions/range"
-              }
-            },
-            "unit": {
-              "type": "integer",
-              "maximum": 255,
-              "minimum": 0
-            }
-          },
-          "required": ["name", "type"]
-        },
-        {
-          "properties": {
-            "type": {
-              "enum": ["string"]
-            },
-            "name": {
-              "$ref": "#/definitions/name"
-            },
-            "max_size": {
-              "type": "integer",
-              "minimum": 0,
-              "exclusiveMinimum": true
-            },
-            "min_size": {
-              "type": "integer",
-              "minimum": 0
-            }
-          },
-          "required": ["name", "type"]
+    "integerType": {
+      "description": "A signed or unsigned integer, can have an optional
+                      prefix, unit, and range.",
+      "properties": {
+        "name": { "$ref": "#/definitions/nameType" },
+        "type": {
+          "enum": [
+            "int8",
+            "int16",
+            "int32",
+            "int64",
+            "uint8",
+            "uint16",
+            "uint32",
+            "uint64"
+          ]
         },
-        {
-          "properties": {
-            "type": {
-              "enum": ["ipv4"]
-            },
-            "name": {
-              "$ref": "#/definitions/name"
-            }
-          },
-          "required": ["name", "type"]
+        "labels": {
+          "description": "A list of labels that name special values.",
+          "type": "array",
+          "items": { "$ref": "#/definitions/labeledIntegerType" },
+          "uniqueItems": true
         },
-        {
-          "properties": {
-            "type": {
-              "enum": ["mac"]
-            },
-            "name": {
-              "$ref": "#/definitions/name"
-            }
-          },
-          "required": ["name", "type"]
+        "prefix": {
+          "description": "The unit prefix, defined in Table A-14 of E1.20.",
```

Comment:
I'm not quite sure why @nomis52 went down this route with his original proposal, OLA stores the prefix as the actual exponent which to me seems a lot more sensible, given you have to do lots of mapping to turn other values here into their equivalent tables in E1.20, this one doesn't seem too bad, we could either just have it as a signed integer or restrict it to only the prefixes listed in A-14 if we wanted to be a tighter match, see:
https://github.com/OpenLightingProject/ola/blob/master/common/rdm/Pids.proto#L66